crates/telio-wg/src/adapter/boring.rs

```diff
@@ -36,7 +36,18 @@ impl BoringTun {
         firewall_reset_connections_callback: super::FirewallResetConnsCb,
     ) -> Result<Self, AdapterError> {
         let config = DeviceConfig {
-            n_threads: 4,
+            // Apple's Boringtun device runs most efficiently on a single perf-core
+            n_threads: {
+                if cfg!(not(any(
+                    target_os = "ios",
+                    target_os = "macos",
+                    target_os = "tvos"
+                ))) {
+                    num_cpus::get()
+                } else {
+                    1
+                }
```

Comment:
Do we have some guarantee that it will be run on the performance core if you have 1 here?

Reply:
That sounds good - but on the other hand it seems that you prepared boringtun for using multiple logical threads on this high performance core - maybe it's worth trying setting that to some higher number?